src/seccomp/src/syscall_table.rs

```diff
@@ -0,0 +1,673 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+/// Creates and owns a mapping from the arch-specific syscall name to the right number.
+#[derive(Debug)]
+pub(crate) struct SyscallTable {
+    map: HashMap<String, i64>,
+    arch: String,
+}
+
+impl SyscallTable {
+    pub fn new(arch: String) -> Self {
+        let mut instance = Self {
+            arch,
+            map: HashMap::with_capacity(332), // nr of syscalls for x86_64 (upper bound)
+        };
+
+        instance.populate_map();
+
+        instance
+    }
+
+    /// Returns the arch-specific syscall number based on the given name.
+    /// Adheres to the "SYS_*" notation.
+    pub fn get_syscall_nr(&self, sys_name: &str) -> Option<i64> {
+        self.map.get(sys_name).copied()
+    }
+
+    // We need to create an architecture-specific map since the set of
+    // possible syscalls is architecture-dependent. We can't just map to their
+    // corresponding libc constant.
+    fn populate_map(&mut self) {
+        if self.arch == "aarch64" {
```

Comment:
Can we  move these into separate files and just include them here?. It would make it easier to manage.

Reply:
makes sense

Reply:
done